### ISSUE TITLE
[ENH] Add testmon

### ISSUE THREAD

**Issue by dainelli98** [sktime/sktime#3993]
Tests take too long to run locally and recently pytest-testmon has added compatibility with pytest-xdist allowing to run only tests affected by code changes while keeping the advantages of being able to run them in parallel.

The solution could be adding pytest-testmon to testing setup.

**Comment by fkiraly** [comment]
good idea, @dainelli98!
but duplicate of https://github.com/sktime/sktime/issues/2530 where some issues have been identified in interaction with how `sktime` collects estimators for its test suite, see there - @tarpas himself has been diving into the issue (but then got busy with other things)

**Comment by fkiraly** [comment]
(any suggestions/updates please in the original issue)

**Comment by fkiraly** [comment]
PS: to reduce test time for one new estimator, you can also use `check_estimator` directly.
